Home fruit shop

Vladslaf opened the pull request.

dopalex commented on src/main/java/storage/Storage.java:
```diff
@@ -2,9 +2,9 @@
 
 import java.util.HashMap;
 import java.util.Map;
-import model.Product;
```
rename папку storage на db

KseniiaMakarova commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,46 @@
+        for (String string : readData) {
+            Transaction transaction = parserService.parse(string);
+            OperationHandler operationHandler = handlersMap.get(transaction.getOperation());
+            operationHandler.operation(transaction.getProduct(), transaction.getQuantity());
```
```suggestion
            operationHandler.operate/process(transaction.getProduct(), transaction.getQuantity());
```

KseniiaMakarova commented on src/main/java/db/Storage.java:
```diff
@@ -0,0 +1,10 @@
+import model.Fruit;
+
+public class Storage {
+    public static final Map<Fruit, Integer> STORAGE = new HashMap<>() {
+    };
```
```suggestion
    public static final Map<Fruit, Integer> storage = new HashMap<>();
```

KseniiaMakarova commented on src/main/java/model/Fruit.java:
```diff
@@ -0,0 +1,37 @@
+package model;
+
+public final class Fruit {
+    private final String nameOfFruit;
```
```suggestion
    private final String name;
```

KseniiaMakarova commented on src/main/java/model/Transaction.java:
```diff
@@ -0,0 +1,48 @@
+        this.operation = operation;
+    }
+
+    public Fruit getProduct() {
+        return fruit;
+    }
+
+    public void setProduct(Fruit fruit) {
+        this.fruit = fruit;
+    }
```
```suggestion
    public Fruit getFruit() {
        return fruit;
    }

    public void setFruit(Fruit fruit) {
        this.fruit = fruit;
    }
```

KseniiaMakarova commented on src/main/java/model/Transaction.java:
```diff
@@ -0,0 +1,48 @@
+        this.quantity = quantity;
+    }
+
+    public enum Operation {
```
мы юзаем этот енам?

KseniiaMakarova commented on src/main/java/service/impl/ReaderServiceImpl.java:
```diff
@@ -0,0 +1,25 @@
+                result.add(row);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to read file");
```
add filename and original exception

KseniiaMakarova commented on src/main/java/strategy/PurchaseOperationHandler.java:
```diff
@@ -0,0 +1,18 @@
+    public void operation(Fruit fruit, Integer quantity) {
+        Integer initialQuantity = Storage.STORAGE.get(fruit);
+        if (Storage.STORAGE.get(fruit) == null) {
+            throw new RuntimeException("Null");
```
let's make the message more informative

Vladslaf commented:
renamed method in OperationHandler class
variable and methods got renamed in Transaction class
removeed unused enum 
added filename and original esception in ReaderServiceImpl class
added informative message in exception in PurchaiseOperationHandler class
